sqe-http-api/Helpers/HttpExceptionMiddleware.cs

```diff
@@ -0,0 +1,40 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using SQE.SqeHttpApi.DataAccess.Helpers;
+
+namespace SQE.SqeHttpApi.Server.Helpers
+{
+    public static class ApplicationBuilderExtensions
+    {
+        public static IApplicationBuilder UseHttpException(this IApplicationBuilder application)
+        {
+            return application.UseMiddleware<HttpExceptionMiddleware>();
+        }
+    }
+
+    internal class HttpExceptionMiddleware
+    {
+        private readonly RequestDelegate next;
+
+        public HttpExceptionMiddleware(RequestDelegate next)
+        {
+            this.next = next;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                await this.next.Invoke(context);
+            }
+            catch (HttpException httpException)
+            {
```

Comment:
If we through different exceptions, you can catch all permission exceptions and set the code to 403, all data exceptions and set the code to 409 (or whatever makes sense).

Reply:
Yeah, I wrote a little middleware that catches these exceptions, reads the HTTP status code, and then packages the exception as that type of HTTP response to be sent back to the user.  There will be a similar component in the SignalR code that packages these exceptions in an appropriate way for the realtime users.